src/stan_math_backend/Transform_Mir.ml

```diff
@@ -308,14 +388,59 @@ let trans_prog (p : typed_prog) =
         true
     | _ -> false
   in
+  let translate_to_open_cl stmts =
+    if !use_opencl then
+      let data_var_idents = List.map ~f:fst p.input_vars in
+      let rec trans_stmt_to_opencl s =
+        { s with
+          stmt=
+            map_statement
+              (switch_expr_to_opencl data_var_idents)
+              trans_stmt_to_opencl s.stmt }
+      in
+      List.map stmts ~f:trans_stmt_to_opencl
+    else stmts
+  in
   let gq =
     ( add_reads p.generate_quantities p.output_vars param_read
+    |> translate_to_open_cl
     |> constrain_in_params p.output_vars
     |> insert_before tparam_start param_writes
     |> insert_before gq_start tparam_writes )
     @ gq_writes
     |> add_fills
   in
+  let log_prob =
```

Comment:
Yeah, here's the bad merge - this log_prob is the one I think we / you moved to, but the one down on line 447 still exists. So it's adding reads twice.

Reply:
Argh. Let me look into that. Sorry :/